src/Microsoft.SqlTools.ServiceLayer/EditData/EditSession.cs

```diff
@@ -378,5 +485,26 @@ public EditUpdateCellResult UpdateCell(long rowId, int columnId, string newValue
                 await errorHandler(e);
             }
         }
+
+        private string ConstructInitializeQuery()
+        {
+            // Using the columns we know, put together a query for the rows in the table
+            var columns = objectMetadata.Columns.Select(col => col.EscapedName);
+            var columnClause = string.Join(", ", columns);
+
+            return $"SELECT ${columnClause} FROM ${objectMetadata.EscapedMultipartName}";
+        }
+
+        public class EditSessionQueryExecutionState
```

Comment:
I generally think of nested classes as being only for the benefit of the class its nested in. There's no use for this class outside of an EditSession, so I think it makes sense for it to be a nested class. If you feel strongly about this, I can change it.

As for documenting, I'll definitely do that